### PR TITLE
CustomResourcePositionForFocusedView

### PR DESCRIPTION
**Pull request by Muhammedguven**
<br>

When we give an custom arrow, the arrowhead doesn't exactly match the focused view. Because in the library, we have a variable as val `arrowHalfWidth = (15 * density) ` 15 means the default drawable resource width value. So its not developed as generic usages. In screenshots, I used custom drawable with width value is 10dp. So, we figure out that there is an problem in here. So we fix the problem with ` val imageStartMargin = tooltipViewState.arrowMargin - (drawable.intrinsicWidth) / 2` usage.

### CODE REVIEW

File: library/src/main/java/com/trendyol/showcase/ui/tooltip/TooltipView.kt [hunk 1]
```diff
@@ -72,8 +72,15 @@ class TooltipView @JvmOverloads constructor(
...
                 visibility = tooltipViewState.getTopArrowVisibility()
                 setTint(tooltipViewState.getBackgroundColor())
-                layoutMarginStart(tooltipViewState.arrowMargin, tooltipViewState.getArrowPercentage())
-                setImageDrawable(ContextCompat.getDrawable(context, tooltipViewState.getTopArrowResource()))
+                val drawable = requireNotNull(
+                    ContextCompat.getDrawable(
+                        context,
+                        tooltipViewState.getTopArrowResource()
+                    )
+                )
```
Comment (bilgehankalkan) [inline library/src/main/java/com/trendyol/showcase/ui/tooltip/TooltipView.kt:80]: Why did you need this? `AppCompatImageView#setImageDrawable` accepts nullable `Drawable`s.

File: library/src/main/java/com/trendyol/showcase/ui/tooltip/TooltipView.kt [hunk 2]
```diff
@@ -72,8 +72,15 @@ class TooltipView @JvmOverloads constructor(
...
+                val imageStartMargin = tooltipViewState.arrowMargin - (drawable.intrinsicWidth) / 2
```
Comment (bilgehankalkan) [inline library/src/main/java/com/trendyol/showcase/ui/tooltip/TooltipView.kt:81]: This operation can be done in `TooltipViewState` or `TooltipFieldUtil` like before.

File: library/src/main/java/com/trendyol/showcase/ui/tooltip/TooltipView.kt [hunk 3]
```diff
@@ -91,8 +98,15 @@ class TooltipView @JvmOverloads constructor(
...
                 setTint(tooltipViewState.getBackgroundColor())
-                layoutMarginStart(tooltipViewState.arrowMargin, tooltipViewState.getArrowPercentage())
-                setImageDrawable(ContextCompat.getDrawable(context, tooltipViewState.getBottomArrowResource()))
+                val drawable = requireNotNull(
+                    ContextCompat.getDrawable(
+                        context,
+                        tooltipViewState.getBottomArrowResource()
+                    )
+                )
+                val imageStartMargin = tooltipViewState.arrowMargin - (drawable.intrinsicWidth) / 2
```
Comment (bilgehankalkan) [inline library/src/main/java/com/trendyol/showcase/ui/tooltip/TooltipView.kt:107]: Same as above.

File: library/src/main/java/com/trendyol/showcase/ui/tooltip/TooltipView.kt (hunk 2 above)
Comment (Muhammedguven) [reply library/src/main/java/com/trendyol/showcase/ui/tooltip/TooltipView.kt]: done e6fb736bec5dd4b51d46aedc39ca0d1b8731e7fa

File: library/src/main/java/com/trendyol/showcase/ui/tooltip/TooltipView.kt (hunk 1 above)
Comment (Muhammedguven) [reply library/src/main/java/com/trendyol/showcase/ui/tooltip/TooltipView.kt]: It's highly unlikely that it will return null. However, in the rare case that it does, the `intrinsicWidth` behaves unexpectedly. Therefore, we intentionally want to trigger a crash in order to catch and handle this situation.

**Review by MertNYuksel — APPROVED** [review]

**Review by bilgehankalkan — APPROVED** [review]
